### PR TITLE
fix 2 label profiling issue

### PR DESCRIPTION
**Pull request by doncarlos999**
There is an issue with this block if you only have 2 labels in your dataset.
See this project: https://studio.edgeimpulse.com/studio/228301/

Current version shows correct validation accuracy in logs but then profiling gives wrong metrics due to sklearn only outputting one set of coeeficients and bias for a binary classification problem. We need 2 for downstream tflite model to function properly.

Current:
Screenshot 2023-05-19 at 10 15 48

After fix:
Screenshot 2023-05-19 at 10 16 50

### CODE REVIEW

**Comment by matpalm** [comment]
good find! there might also be a version of this function from sklearn that does it explicitly as the binary P(y) but this makes sense

**Review by mathijsb — COMMENTED** [review]
LGTM
